### PR TITLE
Average star rating and updates to Review Components (hooks, review form)

### PR DESCRIPTION
**Pull request by Jesse132**
Please review my changes.

getAverageRating is located in the App component and will be passed down to both reviews and productoverview widgets
progress changes to the review widget is also included.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,14 +26,15 @@
   },
   "devDependencies": {
     "@babel/core": "^7.18.6",
+    "@babel/plugin-proposal-object-rest-spread": "^7.18.6",
```
Comment (poioper) [inline package.json:29]: What is this one for?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (poioper) [inline jest.config.js]: Is this file necessary?

File: specs/unit tests/app.test.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+//This file tests the functionality of App specific functions
+const App = require('../../client/src/components/App.jsx');
```
Comment (poioper) [inline specs/unit tests/app.test.js:2]: Is this working?

**Review by poioper — COMMENTED** [review]
Looks great! I have three questions, mainly regarding the jest, could you please answer them?

File: package.json (hunk 1 above)
Comment (Jesse132) [reply package.json:29]: I did an NPM install for the spread operator for babel, because I'm having some troubles with using it with arrays. The new plugin doesn't seem to help

File: jest.config.js (hunk 2 above)
Comment (Jesse132) [reply jest.config.js]: No it is not, I think it might be from our old pre-modified code. Feel free to ignore the changes

File: specs/unit tests/app.test.js (hunk 3 above)
Comment (Jesse132) [reply specs/unit tests/app.test.js:2]: Yes, it does successfully link to the App component which is then used by react-test-renderer to create a copy of the component.

**Review by poioper — APPROVED** [review]
